src/shutdown/umount.c

```diff
@@ -716,6 +753,32 @@ static int mount_points_list_umount(MountPoint **head, bool *changed, bool last_
                 if (nonunmountable_path(m->path))
                         continue;
 
+                if (keep_moving && m->should_move) {
+                        do {
+                                int r;
+                                _cleanup_free_ char *newpath = NULL;
+                                _cleanup_free_ char *escaped = NULL;
+
+                                r = unit_name_path_escape(m->path, &escaped);
```

Comment:
this cannot work. PATH_MAX is 4K, but NAME_MAX is 256.

This low-level stuff (unike the high-level units) really should work for all paths, hence you need to find a different way. (for example, just hash the stuff to a sufficiently large hash value. we have siphas and sha256 anyway9

Reply:
OK. Do you have suggestion? I can use a random temporary name. But we lose information on what the original mountpoint was if there are errors later.

Reply:
I will use sha256